### ISSUE TITLE
Mview cron indexer doesn't skip duplicated ids. ( > 1000 updates )

### ISSUE THREAD

**Issue by srepn** [magento/magento2#35072]
### Preconditions (*)

1. Magento 2.4.3-p1
2. PHP 7.4.26
3. MySQL 8.0.26

### Steps to reproduce (*)

1. Update product more than 1000 times ( default batch size = 1000)
2. check indexer status: `bin/magento indexer:status` // Output: 1 product in backlog
3. Count indexer action executions.

### Expected result (*)

1. Indexer action executed only once.

### Actual result (*)

1. Indexer action executed more than one time

If you have a heavy indexer and a lot of updates the behavior reduces the performance too much.

The method `\Magento\Framework\Mview\View::executeAction` seems like doesn't skip duplicates if there are updates more than the batch size.

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [x] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-November** [comment]
Hi @sergiirepin ,
Thank you for reporting and collaboration. Verified the issue on Magento 2.4-develop branch but could not able to reproduce the issue with below steps:
1. In app/etc/env.php file, changed default batch size to 3 (Since product update for 1000 times is tedious)
```
 'indexer' => [
    'batch_size' => [
        'cataloginventory_stock' => [
            'simple' => 3
        ],
        'catalog_category_product' => 3,
        'catalogsearch_fulltext' => [
            'partial_reindex' => 3,
            'mysql_get' => 3,
            'elastic_save' => 3
        ],
        'catalog_product_price' => [
            'simple' => 3,
            'default' => 3,
            'configurable' => 3
        ],
        'catalogpermissions_category' => 3,
        'inventory' => [
            'simple' => 3,
            'default' => 3,
            'configurable' => 3
        ]
    ]
]
```
2. Updated the same product price and qty from Magento admin panel for more than 3 times.
3. Ran php bin/magento indexer:status
![image](https://user-images.githubusercontent.com/60198231/180425237-a0514c09-f0c5-44f0-a77b-010e211eb085.png)
4. Ran php/bin magento indexer:reindex
Result is it is successfully re indexed for one time.
Kindly recheck the issue on Magento 2.4-develop branch and elaborate steps if the issue is still reproducible.
Thank you.

**Comment by engcom-November** [comment]
Hi @sergiirepin ,
We are closing this issue as there has been no latest update on the same. Kindly reopen / create new issue if you are still facing any issues.
Thank you.
